src/transforms/pipelines/config.rs

```diff
@@ -41,62 +42,115 @@ impl Clone for PipelineConfig {
     }
 }
 
+#[async_trait::async_trait]
+#[typetag::serde(name = "pipeline")]
+impl TransformConfig for PipelineConfig {
+    async fn build(&self, ctx: &TransformContext) -> crate::Result<Transform> {
+        let condition = if let Some(config) = &self.filter {
+            Some(config.build(&ctx.enrichment_tables)?)
+        } else {
+            None
+        };
+
+        let mut transforms = Vec::new();
+        for config in &self.transforms {
+            let transform = match config.build(ctx).await? {
```

Comment:
Since we're no longer expanding to separate transforms, we lose a bit of validation here (e.g. making sure input/output types of connected transforms are compatible).

At a very high level, it would be great to decouple that type of validation from the physical task boundaries. You could imagine a future where we replace transform expansion with the ability to return different types of compound transforms directly from `TransformConfig::build`, and those know how to play nice with a subsequent graph-based validation scheme.

In the meantime, it's likely(?) worthwhile to reimplement some of that validation here, at least to the degree that we think it will be helpful for this transform (if everything is always logs, maybe there's no need).